modules/affinity/src/main/java/org/apache/ignite/internal/affinity/ditributed/AffinityManager.java

```diff
@@ -101,22 +95,22 @@ public AffinityManager(
                             UUID tblId = UUID.fromString(placeholderValue);
 
                             try {
-                                String name = new String(metaStorageMgr.service().get(
+                                String name = new String(metaStorageMgr.get(
                                     new Key(INTERNAL_PREFIX + tblId.toString())).get()
                                     .value(), StandardCharsets.UTF_8);
 
-                                int partitions = configurationMgr.configurationRegistry().getConfiguration(DistributedTableConfiguration.KEY)
+                                int partitions = configurationMgr.configurationRegistry().getConfiguration(TablesConfiguration.KEY)
                                     .tables().get(name).partitions().value();
-                                int backups = configurationMgr.configurationRegistry().getConfiguration(DistributedTableConfiguration.KEY)
-                                    .tables().get(name).backups().value();
-
-                                metaStorageMgr.service().put(evt.newEntry().key(), SerializationUtils.toBytes(
-                                    new RendezvousAffinityFunction(
+                                int replicas = configurationMgr.configurationRegistry().getConfiguration(TablesConfiguration.KEY)
+                                    .tables().get(name).replicas().value();
+
+                                metaStorageMgr.put(evt.newEntry().key(), SerializationUtils.toBytes(
+                                    RendezvousAffinityFunction.assignPartitions(
+                                        baselineMgr.nodes(),
+                                        partitions,
+                                        replicas,
                                         false,
-                                        partitions
-                                    ).assignPartitions(
-                                        networkCluster.allMembers(),
-                                        backups
+                                        null
```

Comment:
At the end of successful watch processing it's required to update Vault configuration and promote it's applied revision. Otherwise you will re-watch all the changes from the start of the world.

Reply:
All get in a watch are invoked with appropriate revision.